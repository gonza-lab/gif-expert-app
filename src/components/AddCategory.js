import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
  const [inputValue, setIValue] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((prevCat) => [inputValue, ...prevCat]);
      setIValue('');
    }
  };

  const handleChange = (e) => {
    setIValue(e.target.value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
