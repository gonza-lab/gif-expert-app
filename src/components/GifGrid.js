import React /* , { useEffect, useState } */ from 'react';
import { GifGridItem } from './GifGridItem';
import { getGif } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
  const { loading, images } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p className="animate__animated animate__flash">Cargando</p>}
      <div className="gifGrid">
        {images.map((img) => (
          <GifGridItem {...img} key={img.id} />
        ))}
      </div>
    </>
  );
};
