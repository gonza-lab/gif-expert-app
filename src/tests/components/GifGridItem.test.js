import { GifGridItem } from '../../components/GifGridItem';
import React from 'react';
import { shallow } from 'enzyme';

describe('Pruebas en GifGridItem', () => {
  const props = {
    title: 'Hola mundo desde Jest',
    url: 'https://localhost/jest.jpg',
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<GifGridItem {...props} />);
  });

  test('Debe mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe ubicar el title correctamente', () => {
    expect(wrapper.find('p').text().trim()).toBe(props.title);
  });

  test('Debe ubicar la url correctamente', () => {
    expect(wrapper.find('img').prop('src')).toBe(props.url);
    expect(wrapper.find('img').prop('alt')).toBe(props.title);
  });

  test('El componentde debe tener la clase correcta', () => {
    expect(
      wrapper.find('div').prop('className').includes('animate__fadeInRightBig')
    ).toBeTruthy();
  });
});
