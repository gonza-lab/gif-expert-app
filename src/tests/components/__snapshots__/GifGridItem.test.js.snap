// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="animate__animated animate__fadeInRightBig"
>
  <img
    alt="Hola mundo desde Jest"
    src="https://localhost/jest.jpg"
  />
  <p>
    Hola mundo desde Jest
  </p>
</div>
`;
